test-data-generator/src/main/scala/za/co/absa/spline/testdatagen/GenCLI.scala

```diff
@@ -20,8 +20,28 @@ import za.co.absa.spline.common.SplineBuildInfo
 import za.co.absa.spline.producer.model.v1_2.ExecutionPlan
 import za.co.absa.spline.testdatagen.generators.{EventGenerator, Graph}
 
+import scala.util.{Failure, Success, Try}
+
 object GenCLI {
 
+  def validateNumericParam(param: String): Either[String, Unit] = {
+    Try(NumericValue(param)) match {
+      case Failure(exception) => Left(exception.getMessage)
+      case Success(_) => Right()
+    }
+  }
+
+  val numericParamText = "positive integers or patterns like: start-end/step, each element positive integer"
+
+  val validGraphValues: String = GraphType.stringValues.mkString(", ")
+
+  def validateGraphType(param: String): Either[String, Unit] = {
+    GraphType.fromString(param) match {
+      case None => Left(s"Invalid provided graph type. Valid values: ${validGraphValues}")
```

Comment:
nit: unnecessary curly brackets (same in a few other places)
```suggestion
      case None => Left(s"Invalid provided graph type. Valid values: $validGraphValues")
```